File: matmul_bycolumn.m
% Usage: A = matmul_bycolumn(B, C)
%
% Computes the matrix product A = B * C, but instead of 
% just calling B * C, it multiplies B by each column of
% C in sequence.
function A = matmul_bycolumn(B, C)
  A = zeros(size(B,1), size(C,2));
  for j = 1:size(C,2)
    A(:,j) = B * C(:,j);
  end

File: matmul_bycolumn.jl
# Usage: C = matmul_bycolumn(A, B)
#
# Computes the matrix product C = A * B, but instead of 
# just calling A * B, it multiplies A by each column of
# B in sequence.
function matmul_bycolumn{T}(A::Matrix{T}, B::Matrix{T})
    C = Array(T, size(A,1), size(B,2));
    for j = 1:size(C,2)
        C[:,j] = A * B[:,j];
    end
    return C
end